Add PHP 8.2 Runtimes

dhrrgn opened the pull request:
PHP 8.2 releases have started being rolled out. This adds the necessary runtimes to allow devs to start testing their applications.

t-richard commented:
Thanks for opening this, I approved the CI to see how it goes!

dhrrgn commented:
Thanks @t-richard. Looks like it failed installing xdebug for the dev images, which makes sense.

Looks like when the 8.1 runtimes were added the dev images were built with `--target=without_extensions`. Is that still the best option (until xdebug is ready for 8.2)?

t-richard commented:
Yes, we'll have to do that again 👍

dhrrgn commented:
@t-richard I implemented the skipping of extensions differently than was done previously.

Can you approve the CI and se if it passes now. 

I am open to changing how we skip the extension install, but want to make sure everything else works, then we can address any changes wanted. 

Thanks!

dhrrgn commented on runtime/layers/fpm-dev/Dockerfile:
```diff
@@ -1,11 +1,12 @@
 
-RUN pecl install xdebug
-RUN cp $(php -r "echo ini_get('extension_dir');")/xdebug.so /tmp
+COPY extensions.sh /tmp/extensions.sh
+RUN chmod +x /tmp/extensions.sh
+RUN /tmp/extensions.sh && rm /tmp/extensions.sh 
```
Moving this into a separate script makes more sense than adding another stage IMO.

Using a separate bash script is also more maintainable and flexible in this scenario IMO.

mnapoli commented:
Thank you!

I'd rather avoid having a new script file (the build is already very complex), could we just skip building `php-82-fpm-dev` for now instead?

dhrrgn commented:
@mnapoli Works for me. I have reverted that change and commented out all references to the `php-82-fpm-dev` build.

GrahamCampbell commented on docs/runtimes/README.md:
```diff
@@ -71,17 +71,19 @@ functions:
 - `${bref:layer.php-80}`
 - `${bref:layer.php-74}`
 - `${bref:layer.php-73}`
+- `${bref:layer.php-82-fpm}`
```
We should probably not include this, or add a note saying coming soon.

dhrrgn replied:
Ah, shoot. Missed that one. I'll update after the CI run. 

How about making it:

```markdown
- `${bref:layer.php-82-fpm}` **experimental**
```

(and the php-82 layer above as well)

t-richard commented on runtime/base/php-82.Dockerfile:
```diff
@@ -0,0 +1,139 @@
+    make install
+
+
+ENV VERSION_PHP=8.2.0alpha2
```
Alpha 3 has been released yesterday, could you update it here ?

dhrrgn commented:
@t-richard Done. I also updated the download URL, as it changed. 

Also, I added the `**experimental**` note in the docs. 

Let me know if there's anything else you need me to do. Thanks!

mnapoli commented:
Let's gooooo!

Thank you @dhrrgn

GrahamCampbell commented:
The download URL typically changes for every release or every other release, for pre-releases. We probably should make the download URL part of the env variable, rather than just the version.

mnapoli commented:
It's not a big deal to change the URL in the Dockerfile. I'd rather avoid complicating the build any further.
